commands/cloudapi-v5/backupunit.go

```diff
@@ -278,19 +287,63 @@ func RunBackupUnitUpdate(c *core.CommandConfig) error {
 }
 
 func RunBackupUnitDelete(c *core.CommandConfig) error {
-	if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete backup unit"); err != nil {
-		return err
-	}
-	c.Printer.Verbose("Backup unit with id: %v is deleting...", viper.GetString(core.GetFlagName(c.NS, cloudapiv5.ArgBackupUnitId)))
-	resp, err := c.CloudApiV5Services.BackupUnit().Delete(viper.GetString(core.GetFlagName(c.NS, cloudapiv5.ArgBackupUnitId)))
-	if resp != nil {
-		c.Printer.Verbose(config.RequestTimeMessage, resp.RequestTime)
-	}
-	if err != nil {
-		return err
-	}
-	if err = utils.WaitForRequest(c, waiter.RequestInterrogator, printer.GetId(resp)); err != nil {
-		return err
+	var resp *resources.Response
+	var err error
+	var backupUnits resources.BackupUnits
+	allFlag := viper.GetBool(core.GetFlagName(c.NS, cloudapiv5.ArgAll))
+	if allFlag {
+		fmt.Printf("Backup Unitts to be deleted:\n")
+		backupUnits, resp, err = c.CloudApiV5Services.BackupUnit().List()
+		if err != nil {
+			return err
+		}
+		if backupUnitsItems, ok := backupUnits.GetItemsOk(); ok && backupUnitsItems != nil {
+			for _, backupUnit := range *backupUnitsItems {
+				if id, ok := backupUnit.GetIdOk(); ok && id != nil {
+					fmt.Printf("BackupUnit Id: " + *id)
+				}
+				if properties, ok := backupUnit.GetPropertiesOk(); ok && properties != nil {
+					if name, ok := properties.GetNameOk(); ok && name != nil {
+						fmt.Printf(" BackupUnit Name: " + *name + "\n")
+					}
+				}
+			}
+
+			if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete all the Backup Units"); err != nil {
+				return err
+			}
+			c.Printer.Verbose("Deleting all the BackupUnits...")
+			for _, backupUnit := range *backupUnitsItems {
+				if id, ok := backupUnit.GetIdOk(); ok && id != nil {
+					c.Printer.Verbose("Deleting Backup unit with id: %v...", *id)
+					resp, err = c.CloudApiV5Services.BackupUnit().Delete(*id)
+					if resp != nil {
+						c.Printer.Verbose(config.RequestTimeMessage, resp.RequestTime)
+					}
+					if err != nil {
+						return err
+					}
+					if err = utils.WaitForRequest(c, waiter.RequestInterrogator, printer.GetId(resp)); err != nil {
+						return err
+					}
+				}
+			}
+		}
```

Comment:
Is there a way in which the "delete-all" functionality can be added in a separate function and that function to be called from here based on flag's value? Adding it like this seems to add a new functionality to a function. And alsoooo, a lot of code smells in the Sonar :)